cargo-concordium/src/schema_json.rs

```diff
@@ -327,7 +329,84 @@ pub fn write_bytes_from_json_schema_type<W: Write>(
                 bail!("JSON String required")
             }
         }
+        Type::ULeb128(constraint) => {
+            if let Value::String(string) = json {
+                let biguint = string.parse().context("Could not parse integer.")?;
+                serial_biguint(biguint, *constraint, out).map_err(|_| anyhow!("Failed writing"))
+            } else {
+                bail!("JSON String required")
+            }
+        }
+        Type::ILeb128(constraint) => {
+            if let Value::String(string) = json {
+                let bigint = string.parse().context("Could not parse integer.")?;
+                serial_bigint(bigint, *constraint, out).map_err(|_| anyhow!("Failed writing"))
+            } else {
+                bail!("JSON String required")
+            }
+        }
+        Type::ByteList(size_len) => {
+            if let Value::String(string) = json {
+                let bytes = hex::decode(string).context("Failed to parse hex")?;
+                let len = bytes.len();
+                write_bytes_for_length_of_size(len, size_len, out)?;
+                for value in bytes {
+                    serial!(value, out)?
+                }
+                Ok(())
+            } else {
+                bail!("JSON String required")
+            }
+        }
+        Type::ByteArray(len) => {
+            if let Value::String(string) = json {
+                let bytes = hex::decode(string).context("Failed to parse hex")?;
+                ensure!(*len == bytes.len() as u32, "Mismatching number of bytes");
+                for value in bytes {
+                    serial!(value, out)?;
+                }
+                Ok(())
+            } else {
+                bail!("JSON String required")
+            }
+        }
+    }
+}
+
+fn serial_biguint<W: Write>(bigint: BigUint, constraint: u32, out: &mut W) -> Result<(), W::Err> {
+    let mut value = bigint;
+    for _ in 0..constraint {
+        let mut byte = (value.clone() % BigUint::from(128u8)).to_u8().unwrap(); // Safe to unwrap since we have truncated
```

Comment:
Why is cloning necessary here? Doesn't `%` take references?

Reply:
No this `%` implementation mutates, but I have found a way around the cloning.